pkg/xds/envoy/types.go

```diff
@@ -227,6 +227,10 @@ func (sa ServicesAccumulator) AddBackendRef(backendRef common_api.BackendRef, cl
 			backendRef: backendRef,
 		}
 	}
+	// prioritize backendRef pointing to real resource
+	if backendRef.ReferencesRealObject() && !sa.services[cluster.Service()].backendRef.ReferencesRealObject() {
```

Comment:
Can backendRef here be `nil` and we will get a panic?

Reply:
The `backendRef` argument isn't a pointer

Reply:
Although it _does_ seem like if `sa.services[cluster.Service()].backendRef` is the empty value, we'll return `true`, is that ok?

Reply:
you are right, we will return true, which does not fit here

Reply:
I've updated `ReferencesRealObject` to take this into account